routetable/route_table.go

```diff
@@ -303,7 +331,26 @@ func (r *RouteTable) Apply() error {
 				retries--
 				continue
 			}
-			logCxt.Debug("Synchronised routes on interface")
+			logCxt.Debug("Synchronised L2 routes on interface")
+
+			// Sync L3 routes.
+			err = r.syncRoutesForLink(ifaceName)
+			if err == IfaceNotPresent {
+				logCxt.Info("Interface missing, will retry if it appears.")
+				break
+			} else if err == IfaceDown {
+				logCxt.Info("Interface down, will retry if it goes up.")
+				break
+			} else if err == IfaceGrace {
+				logCxt.Info("Interface in cleanup grace period, will retry after.")
```

Comment:
Yep. This check is used for all route types though, so I think the right fix is a level lower where we actually check if the route is within the grace period - we should only return this error if the route is not a VXLAN route.

Reply:
Not sure how to do this generically since this is shared code - I don't think it's really a _problem_ per-se, just might take a bit longer to cleanup VXLAN routes added by someone else. Any ideas on how to clean this up? Or, is it worth it?

Reply:
If it's tough to do, think we can live with.  Th grace period is something like 30s.